src/main/java/dev/sigstore/rekor/client/RekorResponse.java

```diff
@@ -15,16 +15,68 @@
  */
 package dev.sigstore.rekor.client;
 
+import com.google.common.reflect.TypeToken;
+import com.google.gson.GsonBuilder;
 import java.net.URI;
+import java.util.Map;
+import org.immutables.value.Value;
 
-public class RekorResponse {
-  private final URI entryLocation;
+/**
+ * Representation of a rekor response with the log location, raw log string and parsed log
+ * information.
+ */
+@Value.Immutable
+public interface RekorResponse {
 
-  public RekorResponse(URI entryLocation) {
-    this.entryLocation = entryLocation;
+  /**
+   * Create a RekorResponse from raw http response information.
+   *
+   * <p>A raw http response looks something like:
+   *
+   * <pre>
+   * {
+   *   "dbf7b3f960d0d5853f80dfc968779554a628b44a30a4c8a2084b5bd2f6970085": {  // log uuid
```

Comment:
and this is changing at some point? I thought I overheard a conversation a few weeks back saying this would turn into a value with some kind of ID key.